packages/adminrouter/extra/src/test-harness/tests/test_service.py

```diff
@@ -0,0 +1,473 @@
+        mocker.send_command(endpoint_id='http://127.0.0.2:5050',
+                            func_name='set_frameworks_response',
+                            aux_data=[])
+        mocker.send_command(endpoint_id='http://127.0.0.1:8123',
+                            func_name='set_srv_response',
+                            aux_data={})
+
+        # Check if we can use root Marathon app data to resolve `/service`
+        # location requests:
+        generic_correct_upstream_dest_test(
+            master_ar_process_fastcache,
+            valid_user_header,
+            '/service/nginx-alwaysthere/foo/bar/',
+            "http://127.0.0.1:16000"
+            )
+
+        # Update the application, wait for cache to register the change
+        new_apps = {"apps": [NGINX_APP_ALWAYSTHERE_DIFFERENTPORT, ]}
+        mocker.send_command(endpoint_id='http://127.0.0.1:8080',
+                            func_name='set_apps_response',
+                            aux_data=new_apps)
+        time.sleep(CACHE_UPDATE_DELAY + 1)
+
+        # Check if the location now resolves correctly to the new app port
+        generic_correct_upstream_dest_test(
+            master_ar_process_fastcache,
+            valid_user_header,
+            '/service/nginx-alwaysthere/foo/bar/',
+            "http://127.0.0.15:16001"
+            )
+
+    def test_if_webui_url_is_resolved_using_framework_id(
+            self, master_ar_process_fastcache, mocker, valid_user_header):
+        # Remove the data from MesosDNS and Marathon mocks w.r.t. resolved service
+        mocker.send_command(endpoint_id='http://127.0.0.1:8080',
+                            func_name='set_apps_response',
+                            aux_data={"apps": []})
+        mocker.send_command(endpoint_id='http://127.0.0.1:8123',
+                            func_name='set_srv_response',
+                            aux_data={})
+
+        # Check if we can use Mesos framework data to resolve `/service`
+        # location request using framework ID:
+        generic_correct_upstream_dest_test(
+            master_ar_process_fastcache,
+            valid_user_header,
+            '/service/0f8899bf-a31a-44d5-b1a5-c8c3f7128905-0000/foo/bar/',
+            "http://127.0.0.1:16000"
+            )
+
+        # Update the application, wait for cache to register the change
+        mocker.send_command(endpoint_id='http://127.0.0.2:5050',
+                            func_name='set_frameworks_response',
+                            aux_data=[NGINX_FWRK_ALWAYSTHERE_DIFFERENTPORT])
+        time.sleep(CACHE_UPDATE_DELAY + 1)
+
+        # Check if the location now resolves correctly to the new app port
+        generic_correct_upstream_dest_test(
+            master_ar_process_fastcache,
+            valid_user_header,
+            '/service/0f8899bf-a31a-44d5-b1a5-c8c3f7128905-0000/foo/bar/',
+            "http://127.0.0.15:16001"
+            )
+
+    def test_if_webui_url_is_resolved_using_framework_name(
```

Comment:
It's both - sometimes it is resolved as a framework, sometimes as an app.

Having the same name helps to catch bugs - service routing priority tests try to resolve a service with a given ID and manipulate the data in Mesos&Maratho mocks so that both framework and a task that refer to this ID exist in their JSON outputs. Tests pass only if AR makes the correct routing decision.

I will change it to `scheduler`, should be more neutral.